### PR TITLE
allow deserializer to return NO_VALUE

### PR DESCRIPTION
**Pull request by sjhewitt**
I have the case that our deserializer can sometimes raise an error (due to a class being moved and `pickle` not being able to find the serialized class.) As a workaround, I've overridden `CacheRegion._parse_serialized_from_backend` to catch any exceptions returned by the deserializer. 
However, it would be useful to be able to return NO_VALUE from the deserializer, then I can move the exception catching logic into the deserializer and not have to override the internal method.

### CODE REVIEW

**Comment by zzzeek** [comment]
Hi -

Let's look at your use case but we wouldnt just take an existing variable and repurpose it with a one-liner like this and no tests.   let's open an issue first and talk about what's needed.

**Comment by zzzeek** [comment]
sorry, it has a test.   Still, this is too random of a way of doing this

**Comment by zzzeek** [comment]
here, can we do something like this?  this woudl eliminate any confusion

```diff
diff --git a/dogpile/__init__.py b/dogpile/__init__.py
index 175c208..52b2f14 100644
--- a/dogpile/__init__.py
+++ b/dogpile/__init__.py
@@ -1,4 +1,4 @@
-__version__ = "1.1.9"
+__version__ = "1.2.0"
 
 from .lock import Lock  # noqa
 from .lock import NeedRegenerationException  # noqa
diff --git a/dogpile/cache/api.py b/dogpile/cache/api.py
index 0717d43..2f22baf 100644
--- a/dogpile/cache/api.py
+++ b/dogpile/cache/api.py
@@ -51,6 +51,15 @@ Serializer = Callable[[ValuePayload], bytes]
 Deserializer = Callable[[bytes], ValuePayload]
 
 
+class CantDeserializeException(Exception):
+    """Exception indicating deserialization failed, and that caching
+    should proceed to re-generate a value
+
+    .. versionadded:: 1.2.0
+    
+    """
+
+
 class CacheMutex(abc.ABC):
     """Describes a mutexing object with acquire and release methods.
 
diff --git a/dogpile/cache/region.py b/dogpile/cache/region.py
index ef0dbc4..6f4e9cd 100644
--- a/dogpile/cache/region.py
+++ b/dogpile/cache/region.py
@@ -26,6 +26,7 @@ from .api import BackendArguments
 from .api import BackendFormatted
 from .api import CachedValue
 from .api import CacheMutex
+from .api import CantDeserializeException
 from .api import CacheReturnType
 from .api import KeyType
 from .api import MetaDataType
@@ -1219,8 +1220,12 @@ class CacheRegion:
 
         bytes_metadata, _, bytes_payload = byte_value.partition(b"|")
         metadata = json.loads(bytes_metadata)
-        payload = self.deserializer(bytes_payload)
-        return CachedValue(payload, metadata)
+        try:
+            payload = self.deserializer(bytes_payload)
+        except CantDeserializeException:
+            return NO_VALUE
+        else:
+            return CachedValue(payload, metadata)
 
     def _serialize_cached_value_elements(
         self, payload: ValuePayload, metadata: MetaDataType

```

**Comment by sjhewitt** [comment]
> here, can we do something like this? this woudl eliminate any confusion
> 

That would work for me

**Comment by zzzeek** [comment]
I still need you to put it into a PR with the test you have to keep it moving, I have zero time to work on dogpile
